.github/workflows/forge-stable.yaml

```diff
@@ -176,6 +178,7 @@ jobs:
       # This will upgrade from testnet branch to the latest main
       FORGE_TEST_SUITE: compat
       IMAGE_TAG: testnet
+      GIT_SHA: ${{ needs.determine-test-metadata.outputs.IMAGE_TAG }} # this is the git ref to checkout
```

Comment:
meaning: we initially start the cluster at `testnet` and then upgrade to the latest image in the specified branch